File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        sofar = {0: 1}
        sum_ = ans = 0
        for n in nums:
            sum_ += n
            check = sum_%k
            if check in sofar:
                ans += sofar[check]
                sofar[check] += 1
            else:
                sofar[check] = 1
        return ans

File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        vector <int> sofar(k);
        sofar[0] = 1;
        long sum_ = 0, ans = 0, mod;
        for(auto n: nums){
            sum_ += n;
            mod = (sum_%k + k)%k;
            ans += sofar[mod]++;
        }
        return ans;
    }
};